lib/logflare/context_cache.ex

```diff
@@ -37,29 +37,23 @@ defmodule Logflare.ContextCache do
     end
   end
 
-  def bust_keys(context, id) do
-    context_cache = cache_name(context)
-
+  def bust_keys(values) do
     {:ok, keys} = Cachex.keys(@cache)
 
-    Stream.each(keys, fn
-      {{^context, ^id}, cache_key} = key ->
-        case Cachex.del(context_cache, cache_key) do
-          {:ok, true} ->
-            Cachex.del(@cache, key)
-
-          {:ok, nil} ->
-            :noop
-        end
-
-      _key ->
-        :noop
+    Enum.each(keys, fn {token, cache_key} = key ->
```

Comment:
Let's add in some telemetry here to track the number of keys and exec duration, if it is large then Task.async_stream or Flow might help.

Reply:
That will be resolved in next PR. We discussed that on Slack with @chasers already.